### PR TITLE
Fixes wrong max slices in GHA

### PR DESCRIPTION
**Pull request by pSchlarb**
Signed-off-by: pSchlarb <p.schlarb@esatus.com>

### CODE REVIEW

**Comment by udosson** [comment]
@pSchlarb thanks for the PR. Could you please provide some context and justify the proposed changes?

**Comment by pSchlarb** [comment]
Yes, shimizut in the Rocketchat mentioned that the tests run by the GHA are only executing 11 of 22 slices per module. 
It seems that the variable used `strategy.job-total` is the sum of all slices across all modules.
So to correctly test all slices with the runner these changes are neccessary.

**Comment by WadeBarnes** [comment]
I don't really like the idea of hard coding the number slices.  The original intent was to be able to update this section and have the test runs automatically adjust accordingly.  Obviously `strategy.job-total` is the number of modules * the number of slices.  We want it to just be the number of slices.  The question is how do we get that value without hard coding it.

```
      matrix:
        module: [indy_node, indy_common]
        slice: [1, 2, 3, 4 ,5, 6, 7, 8, 9, 10, 11]
```

**Comment by WadeBarnes** [comment]
It does not seem to be a way to get the count of items included in the elements of a matrix.

What if we added a `slice_count` environment variable and had it set to the total number of slices, and documented it has to match the number of elements contained in `strategy.matrix.slice`.  It's still a bit of a hack, but better then having to change the value in multiple places is/when the number of elements contained in `strategy.matrix.slice` is changed.

**Comment by pSchlarb** [comment]
I am fine with that but i would call it `SLICE_TOTAL_SLICES`.

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -199,6 +199,8 @@ jobs:
     # - https://github.com/ScaCap/action-surefire-report/issues/17
     env:
       NODE_OPTIONS: '--max_old_space_size=4096'
+      # Total Number of Slices
+      SLICE_TOTAL_SLICES: 11
```
Comment (WadeBarnes) [inline .github/workflows/build.yaml:203]: Please add a note to indicate the value needs to match the total number if items listed in the `strategy.matrix.slice` element.

**Review by WadeBarnes — CHANGES_REQUESTED** [review]

**Comment by WadeBarnes** [comment]
(ci) test this please

**Review by WadeBarnes — APPROVED** [review]

**Pull request merged**